Project.toml

```diff
@@ -27,22 +28,25 @@ CommonSubexpressions = "0.3"
 DiffResults = "1.1"
 DiffRules = "1.4"
 DiffTests = "0.1"
+GPUArraysCore = "0.1, 0.2"
 IrrationalConstants = "0.1, 0.2"
 LogExpFunctions = "0.3"
 NaNMath = "1"
 Preferences = "1"
 SpecialFunctions = "1, 2"
 StaticArrays = "1.5"
-julia = "1.6"
+julia = "1.10"
```

Comment:
When performing this compat bump (IMO would be better in a separate PR but I'm OK with it being included in the PR), we have to do a few more clean ups, e.g., we have to go through the package and remove all unnecessary `VERSION` checks (such as https://github.com/JuliaDiff/ForwardDiff.jl/blob/d90a50581d3224730160fa02358d2546e8a0e18f/src/dual.jl#L303). Probably the reference to Julia 1.6 in https://github.com/JuliaDiff/ForwardDiff.jl/blob/d90a50581d3224730160fa02358d2546e8a0e18f/docs/src/user/advanced.md?plain=1#L89 should be removed as well.